test/features/composer/presentation/composer_controller_test.dart

```diff
@@ -0,0 +1,852 @@
+        when(mockHtmlEditorApi.getText()).thenAnswer((_) async => emailContent);
+          
+        const emailSubject = 'some email subject';
+        composerController?.subjectEmail.value = emailSubject;
+        
+        final toRecipient = EmailAddress('to', 'to@linagora.com');
+        composerController?.listToEmailAddress = [toRecipient];
+
+        final ccRecipient = EmailAddress('cc', 'cc@linagora.com');
+        composerController?.listCcEmailAddress = [ccRecipient];
+
+        final bccRecipient = EmailAddress('bcc', 'bcc@linagora.com');
+        composerController?.listBccEmailAddress = [bccRecipient];
+
+        final identity = Identity();
+        composerController?.identitySelected.value = identity;
+
+        final attachment = Attachment();
+        when(mockUploadController.attachmentsUploaded).thenReturn([attachment]);
+
+        final state = GetAlwaysReadReceiptSettingFailure(Exception());
+
+        final savedEmailDraft = SavedEmailDraft(
+          content: emailContent,
+          subject: emailSubject,
+          toRecipients: {toRecipient},
+          ccRecipients: {ccRecipient},
+          bccRecipients: {bccRecipient},
+          identity: identity,
+          attachments: [attachment],
+          hasReadReceipt: false
+        );
+        
+        // act
+        composerController?.handleFailureViewState(state);
+        await untilCalled(mockHtmlEditorApi.getText());
+        
+        // assert
+        expect(composerController?.savedEmailDraftHash, savedEmailDraft.hashCode);
+      });
+
+      test(
+        'should update _savedEmailDraftHash '
+        'when _initIdentities is called '
+        'and listFromIdentities is not empty '
+        'and selectedIdentity is available',
+      () async {
+        // arrange
+        PlatformInfo.isTestingForWeb = true;
+        const emailContent = 'some email content';
+        composerController?.onChangeTextEditorWeb(emailContent);
+          
+        const emailSubject = 'some email subject';
+        composerController?.subjectEmail.value = emailSubject;
+
+        final toRecipient = EmailAddress('to', 'to@linagora.com');
+        composerController?.listToEmailAddress = [toRecipient];
+
+        final ccRecipient = EmailAddress('cc', 'cc@linagora.com');
+        composerController?.listCcEmailAddress = [ccRecipient];
+
+        final bccRecipient = EmailAddress('bcc', 'bcc@linagora.com');
+        composerController?.listBccEmailAddress = [bccRecipient];
+
+        final selectedIdentity = Identity(id: IdentityId(Id('alice')));
+        when(mockMailboxDashBoardController.composerArguments).thenReturn(
+          ComposerArguments(
+            selectedIdentityId: selectedIdentity.id,
+            identities: [selectedIdentity]));
+
+        final attachment = Attachment();
+        when(mockUploadController.attachmentsUploaded).thenReturn([attachment]);
+
+        grandPermission(Permission.contacts);
```

Comment:
why this test case run in web, we still need to grand permission?

Reply:
Permission condition removed